clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -281,13 +283,16 @@ data NetlistEnv
   -- ^ (Maybe) user given instance/register name
   }
 
+type ComponentMap = OMap Unique ([Bool], SrcSpan, IdentifierSet, Component)
```

Comment:
If you're giving it a name, you _could_ also make it a record :eyes:

Reply:
Yeah good point